torchaudio/compliance/kaldi.py

```diff
@@ -752,141 +753,54 @@ def mfcc(
-    # limit of the function at t = 0
-    weights[t_eq_zero_indices] *= 2 * lowpass_cutoff
-
-    weights /= orig_freq  # size (output_samples_in_unit, max_weight_width)
-    return min_input_index, weights
-
-
-def _lcm(a: int, b: int) -> int:
-    return abs(a * b) // math.gcd(a, b)
-
-
-def _get_num_LR_output_samples(input_num_samp: int,
-                               samp_rate_in: float,
-                               samp_rate_out: float) -> int:
-    r"""Based on LinearResample::GetNumOutputSamples. LinearResample (LR) means that
-    the output signal is at linearly spaced intervals (i.e the output signal has a
-    frequency of ``new_freq``). It uses sinc/bandlimited interpolation to upsample/downsample
-    the signal.
-
-    Args:
-        input_num_samp (int): The number of samples in the input
-        samp_rate_in (float): The original frequency of the signal
-        samp_rate_out (float): The desired frequency
-
-    Returns:
-        int: The number of output samples
-    """
-    # For exact computation, we measure time in "ticks" of 1.0 / tick_freq,
-    # where tick_freq is the least common multiple of samp_rate_in and
-    # samp_rate_out.
-    samp_rate_in = int(samp_rate_in)
-    samp_rate_out = int(samp_rate_out)
-
-    tick_freq = _lcm(samp_rate_in, samp_rate_out)
-    ticks_per_input_period = tick_freq // samp_rate_in
-
-    # work out the number of ticks in the time interval
-    # [ 0, input_num_samp/samp_rate_in ).
-    interval_length_in_ticks = input_num_samp * ticks_per_input_period
-    if interval_length_in_ticks <= 0:
-        return 0
-    ticks_per_output_period = tick_freq // samp_rate_out
-    # Get the last output-sample in the closed interval, i.e. replacing [ ) with
-    # [ ].  Note: integer division rounds down.  See
-    # http://en.wikipedia.org/wiki/Interval_(mathematics) for an explanation of
-    # the notation.
-    last_output_samp = interval_length_in_ticks // ticks_per_output_period
-    # We need the last output-sample in the open interval, so if it takes us to
-    # the end of the interval exactly, subtract one.
-    if last_output_samp * ticks_per_output_period == interval_length_in_ticks:
-        last_output_samp -= 1
-    # First output-sample index is zero, so the number of output samples
-    # is the last output-sample plus one.
-    num_output_samp = last_output_samp + 1
-    return num_output_samp
+def _get_sinc_resample_kernel(orig_freq: int, new_freq: int, lowpass_filter_width: int,
+                              device: torch.device, dtype: torch.dtype):
+    assert lowpass_filter_width > 0
+    kernels = []
+    base_freq = min(orig_freq, new_freq)
+    # This will perform antialiasing filtering by removing the highest frequencies.
+    # At first I thought I only needed this when downsampling, but when upsampling
+    # you will get edge artifacts without this, as the edge is equivalent to zero padding,
+    # which will add high freq artifacts.
+    base_freq *= 0.99
```

Comment:
nit: I'm curious how this constant was chosen, and the description before it doesn't provide that information. Can you elaborate?

Reply:
It was chosen by the previous implementation so I did not try to change it. The previous implementation did not provide a rational on how this was chosen.
In my own implementation for julius I actually used the same one as resampy which is a bit lower (0.945 I think). Overall this constant controls the trade off between keeping high frequencies and removing aliasing. I don't think there is a perfect value. I would say 0.99 is probably not as conservative as the one used in resampy, and can likely introduce some aliasing, but it would be rather subtle.

Reply:
Ah, I missed that the parameter already was being used. Thanks for making it stand out more :)